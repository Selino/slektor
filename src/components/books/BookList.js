import React from "react"
import { Card } from "react-bootstrap"
import { connect } from "react-redux"
import "./books.sass"
import BookSearchForm from "./BookSearchForm"

export const BookList = props => {
  const bookItems = props.books.map(book => (
    <Card key={book.id}>
      <Card.Body>
        {book.volumeInfo.imageLinks !== undefined && (
          <img
            className='book-thumbnail'
            alt={book.volumeInfo.title}
            src={book.volumeInfo.imageLinks.smallThumbnail}
          />
        )}
        <h2>{book.volumeInfo.title}</h2>
        <h4>{book.volumeInfo.authors.map(author => author)}</h4>
        <p>{book.volumeInfo.description}</p>
      </Card.Body>
    </Card>
  ))

  return (
    <div className='book-list'>
      <h3>Book List</h3>
      <BookSearchForm />
      {bookItems}
    </div>
  )
}
const mapStateToProps = state => {
  return {
    books: state.books,
    filters: {
      sortBooksBy: state.filters.sortBooksBy
    }
  }
}

export default connect(mapStateToProps)(BookList)
