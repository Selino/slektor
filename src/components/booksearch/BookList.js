import React from "react"
// import { useSelector, useDispatch } from "react-redux"
import { Card } from "react-bootstrap"
import BookSearchForm from "./BookSearchForm"
// import { sortBooksByAuthor } from "../../actions/FiltersActions"
import { connect } from "react-redux"

function BookList(props) {
  const bookItems = props.books.map(book => (
    <Card key={book.id} style={{ marginBottom: "1rem" }}>
      <Card.Body>
        {book.volumeInfo.imageLinks != undefined && (
          <img
            style={{ float: "left", margin: "0 1rem 1rem 0" }}
            src={book.volumeInfo.imageLinks.smallThumbnail}
          />
        )}
        <h2>{book.volumeInfo.title}</h2>
        <h4>{book.volumeInfo.authors.map(author => author)}</h4>
        <p>{book.volumeInfo.description}</p>
      </Card.Body>
    </Card>
  ))

  return (
    <div>
      <BookSearchForm />
      {bookItems}
    </div>
  )
}

const ConnectedBookList = connect(state => {
  return {
    sortBy: state.filters.sortBooksBy,
    books: state.books
  }
})(BookList)

export default ConnectedBookList

// const [state, setState] = useState({
//   data: {
//     items: []
//   }
// })

// useEffect(() => {
//   getBookList()
// }, [])

// const getBookList = () =>
//   fetch(
//     "https://www.googleapis.com/books/v1/volumes?q=hobbit&maxResults=40&key=AIzaSyANiQq7pX6TFzAv9wuBa6XlzFsaMuNloc4&fields=items(id,volumeInfo(title,description,authors,imageLinks/smallThumbnail))"
//   )
//     .then(res => res.json())
//     .then(data => setState({ data }))
//     .catch(err => {
//       console.log(err)
//     })
// ))
