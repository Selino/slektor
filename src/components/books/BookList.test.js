import React from "react"
import { shallow } from "enzyme"
import { BookList } from "./BookList"
import BooksListData from "../../tests/fixtures/booksFixture"

test("should render the BookList with books", () => {
  const wrapper = shallow(<BookList books={BooksListData} />)
  expect(wrapper).toMatchSnapshot()
})
