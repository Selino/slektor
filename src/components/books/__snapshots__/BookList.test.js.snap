// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the BookList with books 1`] = `
<div
  className="book-list"
>
  <h3>
    Book List
  </h3>
  <Card
    body={false}
    key="pD6arNyKyi8C"
  >
    <CardBody>
      <img
        alt="The Hobbit"
        className="book-thumbnail"
        src="http://books.google.com/books/content?id=pD6arNyKyi8C&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      />
      <h2>
        The Hobbit
      </h2>
      <h4>
        J.R.R. Tolkien
      </h4>
      <p />
    </CardBody>
  </Card>
  <Card
    body={false}
    key="hFfhrCWiLSMC"
  >
    <CardBody>
      <img
        alt="The Hobbit, Or, There and Back Again"
        className="book-thumbnail"
        src="http://books.google.com/books/content?id=hFfhrCWiLSMC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      />
      <h2>
        The Hobbit, Or, There and Back Again
      </h2>
      <h4>
        John Ronald Reuel Tolkien
      </h4>
      <p />
    </CardBody>
  </Card>
  <Card
    body={false}
    key="H8ON-dTgQQYC"
  >
    <CardBody>
      <img
        alt="The Hobbit"
        className="book-thumbnail"
        src="http://books.google.com/books/content?id=H8ON-dTgQQYC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      />
      <h2>
        The Hobbit
      </h2>
      <h4>
        Jude Fisher
      </h4>
      <p />
    </CardBody>
  </Card>
  <Card
    body={false}
    key="KY0BDObXftUC"
  >
    <CardBody>
      <img
        alt="Exploring J.R.R. Tolkien's The Hobbit"
        className="book-thumbnail"
        src="http://books.google.com/books/content?id=KY0BDObXftUC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      />
      <h2>
        Exploring J.R.R. Tolkien's The Hobbit
      </h2>
      <h4>
        Corey Olsen
      </h4>
      <p />
    </CardBody>
  </Card>
  <Card
    body={false}
    key="N_0VhzQKIIAC"
  >
    <CardBody>
      <img
        alt="Finding God in the Hobbit"
        className="book-thumbnail"
        src="http://books.google.com/books/content?id=N_0VhzQKIIAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      />
      <h2>
        Finding God in the Hobbit
      </h2>
      <h4>
        Jim Ware
      </h4>
      <p />
    </CardBody>
  </Card>
</div>
`;
